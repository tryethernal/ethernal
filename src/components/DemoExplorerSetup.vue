<template>
    <v-container fluid>
        <v-row class="primary">
            <v-col cols="12" class="logo-white"><h2>Ethernal</h2></v-col>
        </v-row>
        <v-row justify="center" align="center">
            <v-col cols="12" lg="5">
                <h1>See Ethernal In Action</h1>
                <p class="mb-6">
                    Launch a customized demo explorer, connected to your RPC, displaying your own native token, and get
                    an overview of all of Ethernal's capabilities.
                </p>
                <h2 class="mb-4">
                    What can I expect?
                </h2>
                <ul style="list-style: none;" class="pl-0">
                    <li class="mb-6">
                        <v-icon class="primary--text mr-1">mdi-check</v-icon>
                        Real-time synchronization of blocks & transactions.
                    </li>
                    <li class="mb-6">
                        <v-icon class="primary--text mr-1">mdi-check</v-icon>
                        Processed token transfers & token balance changes for better transaction insights.
                    </li>
                    <li class="mb-6">
                        <v-icon class="primary--text mr-1">mdi-check</v-icon>
                        NFT galleries, with processed metadata (OpenSea compatible).
                    </li>
                    <li  class="mb-6">
                        <v-icon class="primary--text mr-1">mdi-check</v-icon>
                        A link to setup your demo explorer permanently on your account.
                    </li>
                </ul>
            </v-col>
            <v-col cols="12" lg="6">
                <v-card outlined class="primary pa-md-10">
                    <v-card-title class="white--text"><h2>Get Started</h2></v-card-title>
                    <v-card-text>
                        <v-alert v-if="errorMessage" dense text class="white" type="error">{{ errorMessage }}</v-alert>
                        <v-form @submit.prevent="submit" v-model="valid" class="white pa-5">
                            <v-text-field
                                :rules="[v => !!v || 'Name is required']"
                                outlined v-model="name" label="Explorer Name" placeholder="My Explorer Name" class="mb-2" required></v-text-field>
                            <v-text-field
                                :rules="[
                                    v => this.isUrlValid(v) || 'RPC needs to be a valid URL',
                                    v => !!v || 'RPC server is required'
                                ]"
                                outlined v-model="rpcServer" label="RPC URL" placeholder="https://my.rpc.com:8545" class="mb-2" required></v-text-field>
                            <v-text-field outlined v-model="nativeToken" label="Native Token Symbol (optional)" placeholder="ETH" class="mb-2" required></v-text-field>
                            <v-btn large width="100%" :loading="loading" color="primary" :disabled="!valid" type="submit">Generate Explorer</v-btn>
                        </v-form>
                        <v-card outlined class="mt-4" v-if="domain">
                            <v-card-title class="success--text">
                                <v-icon class="success--text mr-2">mdi-check-circle</v-icon>
                                Your demo explorer is ready!
                            </v-card-title>
                            <v-card-text>
                                You can access it here: <a :href="`//${domain}`" target="_blank">https://{{ domain }}</a>
                            </v-card-text>
                        </v-card>
                    </v-card-text>
                </v-card>
            </v-col>
        </v-row>
        <v-divider class="my-10"></v-divider>
        <div align="center">
            <h2 class="mb-10">
                What It Looks Like
            </h2>
        </div>
        <v-row justify="center" align="center" class="mx-md-6">
            <v-col lg="8">
                <video controls preload="metadata" src="../assets/demo_compressed.mp4" type="video/mp4"></video>
            </v-col>
        </v-row>
        <v-divider class="my-10"></v-divider>
        <v-row justify="center" align="center" class="mx-sm-6">
            <v-col cols="12" lg="4">
                <h4 class="mb-2 primary--text">Transactions List</h4>
                <v-img src="../assets/transactions.webp"></v-img>
            </v-col>

            <v-col cols="12" lg="4">
                <h4 class="mb-2 primary--text">Processed Token Transfers</h4>
                <v-img src="../assets/transfers.webp"></v-img>
            </v-col>

            <v-col cols="12" lg="4">
                <h4 class="mb-2 primary--text">Analytics</h4>
                <v-img src="../assets/analytics.webp"></v-img>
            </v-col>
        </v-row>

        <v-row justify="center" align="center" class="mx-md-6">
            <v-col cols="12" lg="4">
                <h4 class="mb-2 primary--text">Contract Interaction</h4>
                <v-img src="../assets/contract.webp"></v-img>
            </v-col>

            <v-col cols="12" lg="4">
                <h4 class="mb-2 primary--text">NFT Gallery</h4>
                <v-img src="../assets/gallery.webp"></v-img>
            </v-col>

            <v-col cols="12" lg="4">
                <h4 class="mb-2 primary--text">Contract Verification</h4>
                <v-img src="../assets/verification.webp"></v-img>
            </v-col>
        </v-row>
        <v-row justify="center" align="center" class="mt-4 mb-10">
            + many other things...
        </v-row>
    </v-container>
</template>

<script>

export default {
    name: 'DemoExplorerSetup',
    data: () => ({
        valid: false,
        name: null,
        rpcServer: null,
        loading: false,
        nativeToken: 'ether',
        errorMessage: null,
        domain: null
    }),
    mounted() {
    },
    methods: {
        submit() {
            this.loading = true;
            this.errorMessage = null;
            this.domain = null;
            this.server.createDemoExplorer(this.name, this.rpcServer, this.nativeToken)
                .then(({ data }) => {
                    this.domain = data.domain;
                })
                .catch(error => {
                    console.log(error)
                    this.errorMessage = error.response && error.response.data || 'Error while creating explorer. Please retry.'
                })
                .finally(() => this.loading = false);
        },
        isUrlValid(url) {
            try {
                new URL(url);
                return true;
            } catch(error) {
                return false;
            }
        },
    }
};
</script>
<style lang="scss">
li {
    display: flex;
}
.v-icon.v-icon {
    align-items: self-start;
}
img {
    border: 1px solid var(--v-primary-base);
    border-radius: 5px;
    height: 100vh;
}
video {
    width: 100%;
}
</style>
