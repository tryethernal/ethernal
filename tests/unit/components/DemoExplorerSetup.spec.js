import MockHelper from '../MockHelper';

import DemoExplorerSetup from '@/components/DemoExplorerSetup.vue';

let helper;
beforeEach(() => {
    jest.clearAllMocks()
    helper = new MockHelper();
});

describe('DemoExplorerSetup.vue', () => {
    it('Should display demo explorer setup page', async () => {
        jest.spyOn(helper.mocks.server, 'getCurrentUser').mockResolvedValue({ data: { id: 1 }});
        const wrapper = helper.mountFn(DemoExplorerSetup);

        expect(wrapper.html()).toMatchSnapshot();
    });
});
