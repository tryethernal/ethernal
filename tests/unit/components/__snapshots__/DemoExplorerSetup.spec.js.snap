// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoExplorerSetup.vue Should display demo explorer setup page 1`] = `
<div class="container container--fluid">
  <div class="row primary">
    <div class="logo-white col col-12">
      <h2>Ethernal</h2>
    </div>
  </div>
  <div class="row align-center justify-center">
    <div class="col-lg-5 col-12">
      <h1>See Ethernal In Action</h1>
      <p class="mb-6">
        Launch a customized demo explorer, connected to your RPC, displaying your own native token, and get
        an overview of all of Ethernal's capabilities.
      </p>
      <h2 class="mb-4">
        What can I expect?
      </h2>
      <ul class="pl-0" style="list-style: none;">
        <li class="mb-6"><i aria-hidden="true" class="v-icon notranslate primary--text mr-1 mdi mdi-check theme--light"></i>
          Real-time synchronization of blocks &amp; transactions.
        </li>
        <li class="mb-6"><i aria-hidden="true" class="v-icon notranslate primary--text mr-1 mdi mdi-check theme--light"></i>
          Processed token transfers &amp; token balance changes for better transaction insights.
        </li>
        <li class="mb-6"><i aria-hidden="true" class="v-icon notranslate primary--text mr-1 mdi mdi-check theme--light"></i>
          NFT galleries, with processed metadata (OpenSea compatible).
        </li>
        <li class="mb-6"><i aria-hidden="true" class="v-icon notranslate primary--text mr-1 mdi mdi-check theme--light"></i>
          A link to setup your demo explorer permanently on your account.
        </li>
      </ul>
    </div>
    <div class="col-lg-6 col-12">
      <div class="primary pa-md-10 v-card v-sheet v-sheet--outlined theme--light">
        <div class="v-card__title white--text">
          <h2>Get Started</h2>
        </div>
        <div class="v-card__text">
          <!---->
          <form novalidate="novalidate" class="v-form white pa-5">
            <div class="v-input mb-2 theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-10" class="v-label theme--light" style="left: 0px; position: absolute;">Explorer Name</label><input required="required" id="input-10" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <div class="v-input mb-2 theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-13" class="v-label theme--light" style="left: 0px; position: absolute;">RPC URL</label><input required="required" id="input-13" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <div class="v-input mb-2 v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-16" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Native Token Symbol (optional)</label><input required="required" id="input-16" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div> <button type="submit" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--large" style="width: 100%;"><span class="v-btn__content">Generate Explorer</span></button>
          </form>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <hr role="separator" aria-orientation="horizontal" class="my-10 v-divider theme--light">
  <div align="center">
    <h2 class="mb-10">
      What It Looks Like
    </h2>
  </div>
  <div class="row mx-md-6 align-center justify-center">
    <div class="col-lg-8 col"><video controls="controls" preload="metadata" src="../assets/demo_compressed.mp4" type="video/mp4"></video></div>
  </div>
  <hr role="separator" aria-orientation="horizontal" class="my-10 v-divider theme--light">
  <div class="row mx-sm-6 align-center justify-center">
    <div class="col-lg-4 col-12">
      <h4 class="mb-2 primary--text">Transactions List</h4>
      <div class="v-image v-responsive theme--light">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
    </div>
    <div class="col-lg-4 col-12">
      <h4 class="mb-2 primary--text">Processed Token Transfers</h4>
      <div class="v-image v-responsive theme--light">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
    </div>
    <div class="col-lg-4 col-12">
      <h4 class="mb-2 primary--text">Analytics</h4>
      <div class="v-image v-responsive theme--light">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
    </div>
  </div>
  <div class="row mx-md-6 align-center justify-center">
    <div class="col-lg-4 col-12">
      <h4 class="mb-2 primary--text">Contract Interaction</h4>
      <div class="v-image v-responsive theme--light">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
    </div>
    <div class="col-lg-4 col-12">
      <h4 class="mb-2 primary--text">NFT Gallery</h4>
      <div class="v-image v-responsive theme--light">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
    </div>
    <div class="col-lg-4 col-12">
      <h4 class="mb-2 primary--text">Contract Verification</h4>
      <div class="v-image v-responsive theme--light">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
    </div>
  </div>
  <div class="row mt-4 mb-10 align-center justify-center">
    + many other things...
  </div>
</div>
`;
